_posts/2020-02-27-integrating-with-a-third-party.md

```diff
@@ -0,0 +1,243 @@
+---
+layout: post
+title: Integrating with a third party
+published: true
+bodyclass: post
+header_type: fixed-dark-header
+author: Dmytro Dashenkov
+---
+
+When developing an [Event-based](https://spine.io/docs/introduction/concepts.html#event) system, it is often tricky to integrate it with other software, be it a third party or a legacy system.
+
+In this article, we will explore the strategies of integrating third-party systems with your Spine-based [Bounded Context](https://spine.io/docs/introduction/concepts.html#bounded-context). 
+
+<!--more-->
+
+Note that we think of a third-party system like of yet another Bounded Context with its own language. The terms “Bounded Context” and “System” are used interchangeably throughout the article.
+
+To start the integration, your team should decide what portion of the other system’s domain model you should adopt and how exactly you want to isolate your system from the third party. In DDD terms, this means choosing the strategy of Context Mapping.
+
+## Context Mapping — TL;DR
+
+For the purpose of this article, let’s consider only the technical (and not the organizational) aspect of a Context Map. You can read more about the concept as a whole in the ["Domain-Driven Design" book](https://dddcommunity.org/book/evans_2003/) by Eric Evans.
+
+A Context Map, from a technical perspective, is the relationship between several Bounded Contexts. The book lists a few archetypes of a Context Map. Some of them are:
+
+ - *Customer/Supplier models*, which establish an upstream-downstream relation. The Supplier team tailors its model specifically for the Customer model. For this strategy to work, the developers of both models have to collaborate.
+ - The *Conformist* pattern, which is very similar to the Customer/Supplier, except that the upstream is not available for change. Thus the downstream has to copy the language into its model thoughtlessly, hence conformism.
+ - *Anticorruption Layer* pattern (a.k.a. *ACL*), describing a Bounded Context, which is not willing to accept the language of the upstream and instead builds an intermediate model in the "no man’s land". The *ACL* translates the language of the upstream into the native language of the downstream Context.
+
+<p class="side-note">
+This list is not exhaustive. The ["Domain-Driven Design" book](https://dddcommunity.org/book/evans_2003/) offers a few more strategies, all worth considering. However, in this article, we are going to describe the listed three patterns, because they are the most commonly used.
+</p>
+
+## The Domain
```

Comment:
As discussed vocally, please have a disclaimer saying that we selected this domain for the vividness of the picture. We do not have experience in creating airport management systems, and our description of the domain will most likely look naïve to the people who work in this area.